### PR TITLE
Add picker localization

### PR DESCRIPTION
**Pull request by imhaage**
## Related Issue
#531

## Description
Make the localization attribute available for the date picker.

## Impacted Areas in Application
List general components of the application that this PR will affect:

* types/index.d.ts
* src/default-props.js
* src/components/m-table-body.js
* src/components/m-table-edit-field.js
* src/components/m-table-edit-row.js
* src/components/m-table-filter-row.js

Small fix on demo.js: `filtering` value was a string instead of a boolean value, there was an error message for invalid proptype.

## Additional Notes
I really love this library, thank you for your great work! I wish I can help to improve it with this contribution.
I had never used TypeScript before, I just added some types for the picker localization. Maybe it's not done right.
I'm a junior developer with only 1 year of experience as a professional (but passionate for years), I wish there won't be too much errors. And if there are some I'll be happy to follow your advices to fix it.

### CODE REVIEW

File: src/components/m-table-edit-field.js [hunk 1]
```diff
@@ -49,6 +49,7 @@ class MTableEditField extends React.Component {
       <MuiPickersUtilsProvider utils={DateFnsUtils}>
         <DatePicker
           {...this.getProps()}
+          {...this.props.localization}
```
Comment (emrekara37) [inline src/components/m-table-edit-field.js]: Thanks,
Can you `{...this.props.localization}` move to `this.getProps()` inside

File: src/components/m-table-edit-field.js [hunk 2]
```diff
@@ -68,6 +69,7 @@ class MTableEditField extends React.Component {
       <MuiPickersUtilsProvider utils={DateFnsUtils}>
         <TimePicker
           {...this.getProps()}
+          {...this.props.localization}
```
Comment (emrekara37) [inline src/components/m-table-edit-field.js]: Thanks,
Can you `{...this.props.localization}` move to `this.getProps()` inside

File: src/components/m-table-edit-field.js [hunk 3]
```diff
@@ -87,6 +89,7 @@ class MTableEditField extends React.Component {
       <MuiPickersUtilsProvider utils={DateFnsUtils}>
         <DateTimePicker
           {...this.getProps()}
+          {...this.props.localization}
```
Comment (emrekara37) [inline src/components/m-table-edit-field.js]: Thanks,
Can you `{...this.props.localization}` move to `this.getProps()` inside

File: src/default-props.js [hunk 4]
```diff
@@ -116,6 +116,12 @@ export const defaultProps = {
         cancelTooltip: 'Cancel',
         deleteText: 'Are you sure delete this row?',
       },
+      picker: {
+        okLabel: 'Ok',
```
Comment (emrekara37) [inline src/default-props.js]: I changed `okLabel:'Yes'` but still saw the 'Ok' text

**Comment by mbrn** [comment]
No response. Closed until contributor's reply.

**Comment by SpearThruster** [comment]
Will we have something equivalent soon?

**Pull request closed without merging**